src/satosa/backends/saml2.py

```diff
@@ -35,6 +35,31 @@
 logger = logging.getLogger(__name__)
 
 
+def get_memorized_idp(context, config, force_authn):
```

Comment:
Tests for the config options can now be written against this function.

Tests for the preference order of the different entity-id options can be written against the `get_idp_entity_id()` function.